azure-mgmt-search/src/main/java/com/microsoft/azure/management/search/SearchService.java

```diff
@@ -0,0 +1,319 @@
+    Refreshable<SearchService>,
+    Updatable<SearchService.Update> {
+
+  /***********************************************************
+   * Getters
+   ***********************************************************/
+
+  /**
+   * Get the hostingMode value.
+   *
+   * @return the hostingMode value
+   */
+  HostingMode hostingMode();
+
+  /**
+   * Get the partitionCount value.
+   *
+   * @return the partitions count of the Search service.
+   */
+  int partitionCount();
+
+  /**
+   * Get the provisioningState value.
+   *
+   * @return the provisioningState value
+   */
+  ProvisioningState provisioningState();
+
+  /**
+   * Get the replicaCount value.
+   *
+   * @return the replicas count of the Search service.
+   */
+  int replicaCount();
+
+  /**
+   * @return the SKU of the Search service.
+   */
+  Sku sku();
+
+  /**
+   * Get the status value.
+   *
+   * @return the status value
+   */
+  SearchServiceStatus status();
+
+  /**
+   * Get the statusDetails value.
+   *
+   * @return the statusDetails value
+   */
+  String statusDetails();
+
+  /**
+   * Gets the primary and secondary admin API keys for the specified Azure Search service.
+   *
+   * @throws IllegalArgumentException thrown if parameters fail the validation
+   * @throws CloudException thrown if the request is rejected by server
+   * @throws RuntimeException all other wrapped checked exceptions if the request fails to be sent
+   * @return the AdminKeys object if successful.
+   */
+  AdminKeys getAdminKeys();
+
+  /**
+   * Gets the primary and secondary admin API keys for the specified Azure Search service.
+   *
+   * @throws IllegalArgumentException thrown if parameters fail the validation
+   * @return the observable to the AdminKeys object
+   */
+  Observable<AdminKeys> getAdminKeysAsync();
+
+  /**
+   * Returns the list of query API keys for the given Azure Search service.
+   *
+   * @throws IllegalArgumentException thrown if parameters fail the validation
+   * @throws CloudException thrown if the request is rejected by server
+   * @throws RuntimeException all other wrapped checked exceptions if the request fails to be sent
+   * @return the List&lt;QueryKey&gt; object if successful.
+   */
+  List<QueryKey> listQueryKeys();
+
+  /**
+   * Returns the list of query API keys for the given Azure Search service.
+   *
+   * @throws IllegalArgumentException thrown if parameters fail the validation
+   * @return the observable to the List&lt;QueryKeyInner&gt; object
+   */
+  Observable<QueryKey> listQueryKeysAsync();
+
+
+  /***********************************************************
+   * Actions
+   ***********************************************************/
+
+  /**
+   * Regenerates either the primary or secondary admin API key. You can only regenerate one key at a time.
+   *
+   * @param keyKind Specifies which key to regenerate. Valid values include 'primary' and 'secondary'.
+   *                Possible values include: 'primary', 'secondary'
```

Comment:
@martinsawicki Assuming this is AutoRest output, I'm assuming this is because AutoRest didn't support documenting individual enum members as of the time that the Search Management Swagger was last updated.